### PR TITLE
interfaces: extra htop apparmor rules

### PR DESCRIPTION
**Pull request by maxiberta**
Fix a number of common denials when running [`htop`](https://snapcraft.io/htop).

`interfaces/builtin/system-observe`:
* Allow read access to `/proc/sys/kernel/sched_autogroup_enabled` and `/proc/*/{,task/*/}autogroup` to [show AGRP (autogroup) and ANI (autogroup nice) columns](https://github.com/htop-dev/htop/pull/722).
* Allow read access to `/proc/*/{,task/*/}io` for [detailed process IO stats](https://github.com/htop-dev/htop/blob/main/linux/LinuxProcessList.c#L512).

`interfaces/builtin/mount-observe`:
* Allow read access to `/proc/1/mounts` to [detect whether it's running under lxc/lxd/docker](https://github.com/htop-dev/htop/pull/967).
* Allow read access to `/proc/spl/kstat/zfs/arcstats` to [calculate RAM usage more accurately](https://github.com/htop-dev/htop/commit/070fe90461182743fabb029415fc1bc59be14f3f).

### CODE REVIEW

**Review by stolowski — APPROVED** [review]
LGTM, thanks!

File: interfaces/builtin/mount_observe.go [hunk 1]
```diff
@@ -39,11 +39,17 @@ const mountObserveConnectedPlugAppArmor = `
...
 @{PROC}/mounts r,
+# Needed by 'htop' to detect whether it's running under lxc/lxd/docker
+@{PROC}/1/mounts r,
+
 owner @{PROC}/@{pid}/mounts r,
 owner @{PROC}/@{pid}/mountinfo r,
 owner @{PROC}/@{pid}/mountstats r,
 /sys/devices/*/block/{,**} r,
 
+# Needed by 'htop' to calculate RAM usage more accurately (and informational purposes, if enabled)
```
Comment (alexmurray) [inline interfaces/builtin/mount_observe.go:50]: Would this be more appropriate under `system-observe` as it doesn't seem mount related to my understanding?

**Review by alexmurray — APPROVED** [review]
LGTM but if the `zfs/arcstats` is about RAM usage that seems like it would be more appropriate under `system-observe` than `mount-observe`.

File: interfaces/builtin/mount_observe.go (hunk 1 above)
Comment (stolowski) [reply interfaces/builtin/mount_observe.go:50]: That's a good point.

**Pull request merged**